pkgs/games/bzflag/default.nix

```diff
@@ -11,10 +11,20 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-nmRlMwK2V72LX5b+EVCp/4Ch1Tptfoo1E4xrGwIAak0=";
   };
 
+  patches = [ ./bzflag-2.4.20-desktop-file.patch ];
```

Comment:
Please us substituteInPlace instead.

Reply:
I'm a bit leery of using substitute given that it fails silently.  If a patch will make things too noisy, can we use sed?

Reply:
substitute does not fail silently, it prints a warning nixpkgs-hammering detects. sed fails fully silent.

> patch will make things too noisy

Its more about repo size.

Reply:
By fail silently I mean whether the build succeeds, not whether the substitution command produces output.  If this attempt to edit the desktop file fails, the build should abort away, like with any other build system.  Otherwise the desktop file will be simply broken.

I realize a PR isn't the place to discuss this, but this also seems to be creating additional work in order to reliably run patched Nixpkgs on a larger scale.  Anyone doing that now has to set up nixpkgs-hammering to avoid unintended patch breakage, rather than being able to rely on build failure?  I don't know anything about nixpkgs-hammering though so please forgive me if I'm missing something.

Reply:
For such simple replaces we are always using substituteInPlace because if we would use patches the repo size would explode. It is very unlikely that the path changes at all and if it does someone will notice it.

Reply:
Sorry, this is a bit too `patch || true` for me.